### ISSUE TITLE
How to completely disable Flash, while enabling YouTube?

### ISSUE THREAD

**Issue by Amstercad** [eXon/videojs-youtube#362]
With a site that we are developing we have encountered an issue that when we try to play a Youtube hosted video under IE 11 on Windows 7, the video will consistently attempt to fall back to Flash, crash, and then use the standard Youtube skin (without controls) rather than the desirable html5 video.js player. We discovered that if we disable flash in the IE browser everything works as expected. Unfortunately, as this will be a public website, that will not be a realistic solution.

Our site uses a mixture of self-hosted MP4 videos, MP3 audio files, and YouTube-hosted videos. To completely disable Flash from being used, at the server-level, I've tried every combination of techOrder possible, along with trying to set the techOrder *settings* without Flash like this:

videojs.options.techOrder = ["html5", "youtube"];

I've also tried everything with and without the techOrder option: "sourceOrder": true

None of my efforts to disable Flash at the server-level have been successful. Note that my efforts also include disabling the video-js.swf file from loading in the first place.

Perhaps an answer to this question might resolve the related VideoJS YouTube plugin issues [#342 (Video not playing Internet Explorer](https://github.com/eXon/videojs-youtube/issues/342)), and [#361 (Not compatible with latest video.js 5.7.0?](https://github.com/eXon/videojs-youtube/issues/361)).

Here are examples of the latest VideoJS YouTube plugin working (at least when using pure html5 browsers) and various versions of VideoJS libraries:

http://5-0-2.videojs-test-xyz.org
http://5-1-0.videojs-test-xyz.org
http://5-2-4.videojs-test-xyz.org
http://5-3-0.videojs-test-xyz.org
http://5-4-6.videojs-test-xyz.org
http://5-5-3.videojs-test-xyz.org
http://5-6-0.videojs-test-xyz.org
http://5-7-1.videojs-test-xyz.org

 And to try these same versions without any influence due to caching:
http://5-0-2.dev.videojs-test-xyz.org
http://5-1-0.dev.videojs-test-xyz.org
http://5-2-4.dev.videojs-test-xyz.org
http://5-3-0.dev.videojs-test-xyz.org
http://5-4-6.dev.videojs-test-xyz.org
http://5-5-3.dev.videojs-test-xyz.org
http://5-6-0.dev.videojs-test-xyz.org
http://5-7-1.dev.videojs-test-xyz.org

The sub-domain URLs indicate which version of the VideoJS library is used, with all else being equal.

What is the appropriate mechanism to prevent Video.js from falling back to flash on the site, to ensure IE only displays using the html5 videojs player, just like other common up-to-date browsers?

**Comment by gkatsev** [comment]
Yeah, I wonder if the issue you're seeing is actually related to the other youtube-tech isn't working on IE issues.
As for how to force flash from loading, theoretically, it should be just removing `Flash` from the tech order. However, I think the youtube tech just loads in whatever youtube wants and so if youtube decided to try and load in flash, it can. There may be some kind of option that can be given to youtube's API to have it only load in html5 content but I'm not sure.

**Comment by Amstercad** [comment]
Since late last week we've noticed that *all* of our code is working, again. Meaning our website versions from last December along with the website versions from February, that prompted me to write this issue in the first.

I thoroughly tested every conceivable iteration of techOrder, with no desirable results. I am positive techOrder had no relevance on this particular issue.

I swear, during December all of the website code using both VideoJS player and the YouTube plugin worked, however in February everything failed, causing me to question my earlier results, and my sanity. In retrospect, now that *everything* works, (all versions of code, old and new), we're squarely pointing the finger at Microsoft Internet Explorer and its 'updates'. I've searched the internet for Internet Explorer Release Notes, and Microsoft doesn't even publish them.

The moral to this sad story is: how much it costs to support Microsoft and its web browsers. The End. This issue is now closed, thank you very much.
